### PR TITLE
FH-4070 Add mcp-mobile-keys configmap on MCP create

### PR DESCRIPTION
**Pull request by aidenkeating**

### CODE REVIEW

File: pkg/mobile/app/service.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+)
+
+type Service struct {
+}
+
+func (s *Service) Create(appCrudder mobile.AppCruder, app *mobile.App) error {
+	uid := uuid.NewV4()
+	app.APIKey = uid.String()
+	switch app.ClientType {
+	case "android":
```
Comment (maleck13) [inline pkg/mobile/app/service.go]: probably could extract these to constants as they are very likely to be used again

File: pkg/web/mobileappHandler.go [hunk 2]
```diff
@@ -99,34 +101,22 @@ func (m *MobileAppHandler) Create(rw http.ResponseWriter, req *http.Request) {
...
+	app.MetaData = map[string]string{}
 
-	if err := appRepo.Create(app); err != nil {
-		err = errors.Wrap(err, "mobile app create: Attempted to create app via app repo")
+	if err := m.appService.Create(appRepo, app); err != nil {
+		err = errors.Wrap(err, "mobile app handler, failed to create app")
 		handleCommonErrorCases(err, rw, m.logger)
 		return
 	}
+	//todo logic is creeping in here should only be for parsing and rendering. Move to mobile package
```
Comment (maleck13) [inline pkg/web/mobileappHandler.go]: can remove this comment now

File: pkg/web/mobileappHandler_test.go [hunk 3]
```diff
@@ -22,11 +24,76 @@ func setupMobileAppHandler(kclient kubernetes.Interface) http.Handler {
...
+			Client: func() kubernetes.Interface {
+				client := &fake.Clientset{}
+				client.AddReactor("get", "configmaps", func(action ktesting.Action) (handled bool, ret runtime.Object, err error) {
+					return true, &v1.ConfigMap{
+						Data: map[string]string{},
+					}, nil
+				})
+				return client
+			},
+			StatusCode: 500,
```
Comment (maleck13) [inline pkg/web/mobileappHandler_test.go]: This should probably be a 400. Is it failing because the clientType is missing?

**Comment by maleck13** [comment]
@aidenkeating couple of comments otherwise good to go

File: pkg/mobile/app/service.go (hunk 1 above)
Comment (aidenkeating) [reply pkg/mobile/app/service.go]: @maleck13 Would the constants stay in this package or move to types.go?

**Comment by aidenkeating** [comment]
@maleck13 Thanks for the review! Addressing these comments now

**Comment by aidenkeating** [comment]
@maleck13 I think this is ready. Would you mind taking a look again?

There have been three commits since the last review:

e4dfcbd
0beb962
ff3606c

The bulk of the changes is moving the ConfigMap creation to the server startup instead of inside the APBs.

**Comment by aidenkeating** [comment]
@maleck13 Like this?

ff3606c

**Pull request merged**